zk/conn.go

```diff
@@ -218,9 +222,12 @@ func Connect(servers []string, sessionTimeout time.Duration, options ...connOpti
 	}
 
 	conn.setTimeouts(int32(sessionTimeout / time.Millisecond))
+	// TODO: this context should become to control the goroutines
+	// and connection itself, like use net.DialContext.
+	ctx := context.Background()
```

Comment:
can we have ctx be based in as a `connOption`?

Reply:
in my prior comment i basically don't want to expose this as an option yet until this is used for actually canceling the connection or operations. This currently is used for testing but will not be used for main cancellation control.

Reply:
i made the withContext a private option would that be ok? this would be used for testing only at this point.

Reply:
offline discussion to update comment to this effect.

Reply:
a private option or just updating TODO to indicate that it will later be passed in seems fine to me.